### ISSUE TITLE
Router.__call__ does not handle interrupted system calls in select

### ISSUE THREAD

**Issue by Ronen2** [selectel/pyxs#9]
I'm integrating pyxs within a twisted base application.
I hit an issue where the select.select() call within Router.__call__ was interrupted.
This call is within a try block without any except clause. It does not handle interrupted system calls within the select system call.
It seems that any error from select.select() would end up with a select.error exception which is not caught and the Router connection is terminated.
Is it right to catch such exception and re-issue the select.select() call?

**Comment by superbobry** [comment]
> Is it right to catch such exception and re-issue the select.select() call?

Yes, that sounds right. I'd welcome a PR with the try-except and a test for this!
